### PR TITLE
[Modelzoo] Creat SimpleMultiTask model.

### PR DESCRIPTION
**Pull request by bwlodarcz**
Implementation of SimpleMultiTask based on EasyRec.

### CODE REVIEW

**Comment by changqi1** [comment]
@bwlodarcz  Please give us the following info:

image

**Comment by changqi1** [comment]
@bwlodarcz  I have 3 questions.
1)	For AUC, we should be more than 0.7, and use Ali-CCP dataset to improve the AUC.
2)	For Globalsteps/sec, why isn't DeepRec w/ oneDNN FP32+BF16 better than DeepRec w/ oneDNN FP32?  We want to get DeepRec w/ oneDNN FP32+BF16 better than DeepRec w/ oneDNN FP32.
3)	Need SPR data.

**Comment by Duyi-Wang** [comment]
@bwlodarcz  acc calculation in line 136 need update.`self.acc, self.acc_op = tf.metrics.accuracy(labels=labels,predictions=preds)`
'tf.metrics.accuracy'  don't accept probability as predictions. 
Please update to `self.acc, self.acc_op = tf.metrics.accuracy(labels=labels,predictions=tf.round(preds))`

**Comment by aalbersk** [comment]
> @bwlodarcz acc calculation in line 136 need update.`self.acc, self.acc_op = tf.metrics.accuracy(labels=labels,predictions=preds)` 'tf.metrics.accuracy' don't accept probability as predictions. Please update to `self.acc, self.acc_op = tf.metrics.accuracy(labels=labels,predictions=tf.round(preds))`

@Duyi-Wang fixed. FYI, I'm currently responsible for SimpleMultiTask.

**Comment by aalbersk** [comment]
@Duyi-Wang ACC/AUC/Globalsteps/s have been improved. Is there anything else that is required for this PR to be merged?

**Comment by Duyi-Wang** [comment]
@aalbersk The perf data is on taobao_dataset or Ali_CCP? If taobao_dataset, please use the Ali_CCP dataset. If Ali_CCP, please update the dataset part in README.

**Comment by kmagiers** [comment]
@Duyi-Wang 
> @aalbersk The perf data is on taobao_dataset or Ali_CCP? If taobao_dataset, please use the Ali_CCP dataset. If Ali_CCP, please update the dataset part in README.

Please do not add requirements on the fly and keep our team to different standards. DSSM already uses this dataset with similar ACC/AUC and is accepted. Please do not block this PR but make separate Jira task for this model to be migrated to different dataset which will be done in separate PR.

**Comment by Duyi-Wang** [comment]
> @Duyi-Wang
> 
> > @aalbersk The perf data is on taobao_dataset or Ali_CCP? If taobao_dataset, please use the Ali_CCP dataset. If Ali_CCP, please update the dataset part in README.
> 
> Please do not add requirements on the fly and keep our team to different standards. DSSM already uses this dataset with similar ACC/AUC and is accepted. Please do not block this PR but make separate Jira task for this model to be migrated to different dataset which will be done in separate PR.

@kmagiers Firstly, our requirements are not added on the fly. Changqing has requested it at the first comment of this PR. And at the sync meeting, I think we have aligned that these models you working on should use Ali-CCP instead of taobao_dataset due to acc/auc issue.
Secondly, DSSM will move to Ali-CCP dataset when your models are ready with this dataset. So with your help, I can quickly finish this dataset migration. This accepted DSSM code is buggy and needs to be improved. It was accepted for time reasons.
Thirdly, it doesn't need to open a new JIRA or PR, because at least one month ago, you have known that we have given up building model on taobao_dataset.
Fourthly, where is your SPR data? Changqing has commented in this PR about one month ago. 
If our requirements are met, the PR can be merged right away.

**Comment by akukushk** [comment]
Hi @Duyi-Wang requirements are evolving since Dec'21 for DeepRec ModelZoo tasks. It is hard to estimate accurately resources and delivery time if new requirements are coming all the time. Creating new JIRAs and PRs for new requirements is a common requirements management and execution tracking practice. Let's discuss the change management process on the Thur sync this week. 

Regarding SimpleMultiTask please can elaborate why we need to move to Ali-CCP dataset since AUC is good with taobao dataset. Probably there was some miscommunication we talked about Ali-CCP dataset for ESMM model only.

SPR machine still doesn't work for us. Can you please help to resolve the issue with the machine.

**Comment by app-on-mic** [comment]
@akukushk Hi Anna, I believe there is some misunderstanding here. The requirements always are kept the same for all of the models, just as you did on ESMM model.  We suppose that Ali-CPP data is more friendly for improving accuracy and hope to adopt it to the models from EasyRec(All of the models in Poland team's tasks are from EasyRec).   
If you think it's better to create a new JIRA to manage and track the status, we could talk about it in the bi-weekly meeting. 

And for SPR platform, we are working hard to fix the issue caused by the new kernel patch, which also has blocked some of our work already.

**Comment by aalbersk** [comment]
SPR numbers for SimpleMultiTask available on jira: WC1-1044

File: modelzoo/SimpleMultiTask/train.py [hunk 1]
```diff
@@ -0,0 +1,563 @@
...
+    test_file = os.path.join(args.data_location, 'taobao_test_data')
+    if not os.path.exists(args.data_location):
+        raise ValueError(f'[ERROR] data location: {args.data_location} does not exist. '
+                         'Please provide valid path')
+
+    if not os.path.exists(train_file) or (not args.no_eval and not os.path.exists(test_file)):
+        raise ValueError('[ERROR] taobao_train_data or taobao_test_data does not exist '
+                         'in the given data_location. Please provide valid path')
+
+    if not os.path.exists(args.output_dir):
```
[the remark this answers is not in the document]
Comment (akukushk) [reply modelzoo/SimpleMultiTask/train.py]: @chenzhiyin you have commented the Outdated code. This code does not exist. s

**Comment by changqi1** [comment]
|  | ACC | |
| -- | -- | -- |
|  | value | percent |
| Community TF | 0.97315 | baseline |
| DeepRec FP32 | 0.97315 | 100.00% |
| DeepRec BF16 | 0.97315 | 100.00% |

|  | AUC | |
| -- | -- | -- |
|  | value | percent |
| Community TF | 0.745045 | baseline |
| DeepRec FP32 | 0.745045 | 100.00% |
| DeepRec BF16 | 0.751338 | 100.84% |

|  | Gsteps/s | |
| -- | -- | -- |
|  | value | percent |
| Community TF | 148.25659 | baseline |
| DeepRec FP32 | 189.32044 | 127.70% |
| DeepRec BF16 | 207.40168 | 139.89% |

**Pull request merged**